normandy/base/utils.py

```diff
@@ -50,7 +50,14 @@ def get_client_ip(request):
 
 
 def filter_m2m(qs, field, values):
-    qs.annotate(_count=Count(field)).filter(_count=len(values))
+    """
+    Filters a queryset by an exact list of many to many relations.
+    """
+    values = list(values)
+
+    qs = qs.annotate(_count=Count(field)).filter(_count=len(values))
+
+    if len(values):
+        qs = qs.filter(**{'{}__in'.format(field): values})
```

Comment:
Does `in` do an exact comparison? I would think that this would match all revisions that have any one of the given values in it. Which, if I'm understanding the docstring, is not what we want.